lecture-notes/3-bld.md

```diff
@@ -680,31 +680,271 @@ $
 
 you can use `HELLO` as a Make variable.
 
-In short, environment variables (e.g. `export CFLAGS=-g ; make hello`):
+In short, when using environment variables in commands such as `export CFLAGS=-g ; make hello`:
 
-* Tells the shell to run make with `CFLAGS` set in the environment
+* The shell runs make with `CFLAGS` set in the environment
 * Make imports all environment variables as Make variables
 * However, assignments in the Makefile override environment variables! `=` and
   `:=` replace them entirely, while `+=` appends to them
 
+So, for example, the following Makefile overrides the environment variable:
+
+```make
+.PHONY: all
+HELLO:=Max
+all:
+	echo $(HELLO)
+```
+
+```console
+$ HELLO=Tom make
+echo Max
+Max
+$
+```
+
 #### Variable overrides
 
-<!-- TODO -->
+Variable overrides are different. While environment variables allow for
+customization from the outside, variable overrides are a feature of Make that
+let the user forcibly change variable values.
+
+This means that making changes to variables need not involve mucking around
+inside a Makefile and finding all places where a variable is initialized.
+Instead, you can decide once and for all what the value should be when invoking
+`make`.
+
+Imagine using someone else's Makefile where they have hard-coded `CC:=gcc` at
+the top. No need to edit; run `make CC=clang` and enjoy using your compiler of
+choice.
 
-In short, variable overrides (e.g. `make CFLAGS=-g hello`):
+In short, when using variable overrides in commands such as `make CFLAGS=-g hello`:
 
-* *Not* an environment variable: value passed as an argument to `make` directly
-* Sets the given Make variable to the given value
+* The shell does *not* use an environment variable: the shell passes the string
```

Comment:
"does not use" is a bit imprecise wording. Maybe say something like "Due to its position after the command name, the shell treats `CFLAGS=-g` as just another argument to pass to Make. In other words, no shell variable ever gets set—Make is what interprets the variable assignment."